### ISSUE TITLE
UIA end of text is defined as last visible line

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#6453]
# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable): master 9ce884c4

Any other software?
```

# Steps to reproduce
This is in ConHost and Terminal.
1. generate output (run `dir` then `dir /?`)
2. using NVDA get the current line
3. scroll up a few lines
4. use NVDA to read the previous line
5. repeat step 4 until you're on the last visible line
6. use NVDA to read the next line

# Expected behavior

Reads the next line.

# Actual behavior

Refuses to read the next line because you are "at the bottom".

**Comment by DHowett** [comment]
Moved from terminal 2.0 to 21H1 because it's a conhost bug.

**Comment by carlos-zamora** [comment]
Just tested this in Terminal. It repro's there too.

**Comment by codeofdusk** [comment]
NVDA [bounds UIA to the visible text](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/UIA/winConsoleUIA.py#L23). Otherwise we'd get thousands upon thousands of blank lines at the end of output.

**Comment by DHowett** [comment]
That’s helpful to know. Since the traditional console can have text that’s below the viewport (whereas terminal cannot), how does the user discover that there is additional content? Is it an uncommon-enough use case that it doesn’t matter much?

**Comment by codeofdusk** [comment]
I've never run into this.

Would it be possible to remove the literally thousands of empty lines from UIA? It slows down [our diffing algorithm](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/behaviors.py#L324) and sometimes I can get lost in all the blanks! If so I could take out the visible range bounding...

**Comment by codeofdusk** [comment]
Maybe this is part of what's causing #5481? (in any case, if we don't have to bound to the visible ranges any more that issue would be mitigated).

**Comment by carlos-zamora** [comment]
I'm actually going to close this. This is specific to NVDA. Narrator, Inspect.exe, and Accessibility Insights don't do this. This is just NVDA bounding UIA to the visible text to improve performance.
